app/assets/javascripts/controllers/tree_view_controller.js

```diff
@@ -16,8 +16,41 @@
         });
         vm.$scope.$apply();
       }
+
+      if (payload.updateTrees && _.isObject(payload.updateTrees) && Object.keys(payload.updateTrees).length > 0) {
+        vm.updateTrees(payload.updateTrees);
+        vm.$scope.$apply();
+      }
     });
 
+    vm.updateTrees = function(data) {
+      if (vm.data[data.tree]) {
+        vm.data[data.tree].forEach(function(node, index) {
+          vm.createTreePath(node, data.key, '[' + index + ']');
+          if(vm.nodePath !== undefined) {
+            Object.keys(data.data).forEach(function(key) {
+              _.set(vm.data, data.tree + vm.nodePath + '.' + key, data.data[key]);
+            });
+            vm.data = angular.copy(vm.data)
+          }
+        });
+      } else {
+        console.warn('Sub tree' + data.tree + ', cloud not be reloaded, because it is not in current tree');
+      }
+    }
+
+    vm.createTreePath = function(node, target, path = '') {
```

Comment:
What this function does? It's not clear from its name and recursion is always scary.

Reply:
It basically goes through the tree and tries to find the node which should be modified based on the node key. Thet it will save the string the the path eg `nodes[0].nodes[3]...`. And that path is the used to change the data.. I will add some comment.